core/cli/src/rc-file.ts

```diff
@@ -1,19 +1,41 @@
-import { cosmiconfig } from 'cosmiconfig'
+import { styles as s } from '@dotcom-tool-kit/logger'
 import { RCFile } from '@dotcom-tool-kit/types/src'
+import { cosmiconfig } from 'cosmiconfig'
+import * as path from 'path'
+import type { Logger } from 'winston'
 
 export const explorer = cosmiconfig('toolkit', { ignoreEmptySearchPlaces: false })
+const emptyConfig = { plugins: [], hooks: {}, options: {} }
+let rootConfig: string | undefined
 
 type RawRCFile = {
   [key in keyof RCFile]?: RCFile[key] | null
 }
 
-export async function loadToolKitRC(root: string): Promise<RCFile> {
-  const result = (await explorer.search(root)) as { config: RawRCFile | null } | null
-  if (!result || !result.config) return { plugins: [], hooks: {}, options: {} }
+export async function loadToolKitRC(logger: Logger, root: string, isAppRoot: boolean): Promise<RCFile> {
+  const result = await explorer.search(root)
+
+  if (!result?.config) {
+    return emptyConfig
+  }
+  if (isAppRoot) {
+    rootConfig = result.filepath
```

Comment:
**question:** What do we do with `rootConfig` here? It isn't referenced later on apart from on line 23 but as that is in an else condition it would have no context of this change… I think.

Reply:
`rootConfig` is stored at the module level so that it can be referenced across multiple calls to `loadToolKitRC`

Reply:
Right of course! This function will be called multiple times in which we store the `rootConfig` in local memory for the subsequent calls.